boards/PDM/PDM.ioc

```diff
@@ -1,37 +1,44 @@
 ADC1.EnableAnalogWatchDog3=false
 ADC1.EnableInjectedConversion=ENABLE
 ADC1.EnableRegularConversion=ENABLE
-ADC1.ExternalTrigConv=ADC_SOFTWARE_START
-ADC1.ExternalTrigConvEdge=ADC_EXTERNALTRIGCONVEDGE_NONE
-ADC1.IPParameters=Channel-IN2,Channel-IN7,Channel-IN8,Channel-IN9,Channel-IN11,Channel-IN12,Channel-IN14,ClockPrescaler,Resolution,DataAlign,ScanConvMode,ContinuousConvMode,DiscontinuousConvMode,DMAContinuousRequests,EOCSelection,Overrun,LowPowerAutoWait,EnableRegularConversion,NbrOfConversion,ExternalTrigConv,ExternalTrigConvEdge,EnableInjectedConversion,InjNumberOfConversion,EnableAnalogWatchDog1,EnableAnalogWatchDog2,EnableAnalogWatchDog3,Rank-7\#ChannelRegularConversion,Channel-7\#ChannelRegularConversion,SamplingTime-7\#ChannelRegularConversion,OffsetNumber-7\#ChannelRegularConversion,Offset-7\#ChannelRegularConversion,SubFamily,master
+ADC1.ExternalTrigConv=ADC_EXTERNALTRIGCONV_T2_TRGO
+ADC1.ExternalTrigConvEdge=ADC_EXTERNALTRIGCONVEDGE_RISING
+ADC1.IPParameters=ClockPrescaler,Resolution,DataAlign,ScanConvMode,ContinuousConvMode,DiscontinuousConvMode,DMAContinuousRequests,EOCSelection,Overrun,LowPowerAutoWait,EnableRegularConversion,NbrOfConversion,ExternalTrigConv,ExternalTrigConvEdge,EnableInjectedConversion,InjNumberOfConversion,EnableAnalogWatchDog1,EnableAnalogWatchDog2,EnableAnalogWatchDog3,Rank-7\#ChannelRegularConversion,Channel-7\#ChannelRegularConversion,SamplingTime-7\#ChannelRegularConversion,OffsetNumber-7\#ChannelRegularConversion,Offset-7\#ChannelRegularConversion,Rank-8\#ChannelRegularConversion,Channel-8\#ChannelRegularConversion,SamplingTime-8\#ChannelRegularConversion,OffsetNumber-8\#ChannelRegularConversion,Offset-8\#ChannelRegularConversion,Rank-9\#ChannelRegularConversion,Channel-9\#ChannelRegularConversion,SamplingTime-9\#ChannelRegularConversion,OffsetNumber-9\#ChannelRegularConversion,Offset-9\#ChannelRegularConversion,Rank-10\#ChannelRegularConversion,Channel-10\#ChannelRegularConversion,SamplingTime-10\#ChannelRegularConversion,OffsetNumber-10\#ChannelRegularConversion,Offset-10\#ChannelRegularConversion,master
 ADC1.InjNumberOfConversion=0
 ADC1.LowPowerAutoWait=DISABLE
-ADC1.NbrOfConversion=1
+ADC1.NbrOfConversion=4
+ADC1.Offset-10\#ChannelRegularConversion=0
 ADC1.Offset-7\#ChannelRegularConversion=0
+ADC1.Offset-8\#ChannelRegularConversion=0
+ADC1.Offset-9\#ChannelRegularConversion=0
+ADC1.OffsetNumber-10\#ChannelRegularConversion=ADC_OFFSET_NONE
 ADC1.OffsetNumber-7\#ChannelRegularConversion=ADC_OFFSET_NONE
+ADC1.OffsetNumber-8\#ChannelRegularConversion=ADC_OFFSET_NONE
+ADC1.OffsetNumber-9\#ChannelRegularConversion=ADC_OFFSET_NONE
 ADC1.Overrun=ADC_OVR_DATA_OVERWRITTEN
+ADC1.Rank-10\#ChannelRegularConversion=4
 ADC1.Rank-7\#ChannelRegularConversion=1
+ADC1.Rank-8\#ChannelRegularConversion=2
+ADC1.Rank-9\#ChannelRegularConversion=3
 ADC1.Resolution=ADC_RESOLUTION_12B
-ADC1.SamplingTime-7\#ChannelRegularConversion=ADC_SAMPLETIME_1CYCLE_5
-ADC1.ScanConvMode=ADC_SCAN_DISABLE
-ADC1.SubFamily=STM32F302xC
```

Comment:
Now that we are using 1kHz interrupts to initiate the conversion we can probs get away with less sample time cycles

![image](https://user-images.githubusercontent.com/32230980/81884268-466fb300-954c-11ea-9c8a-3ea25907115b.png)

Reply:
Sampling rate is dependent on the [external impedance ](https://www.st.com/resource/en/application_note/cd00211314-how-to-get-the-best-adc-accuracy-in-stm32-microcontrollers-stmicroelectronics.pdf#page=19)seen by the adc, the datasheet has a section on calculating Rain. You have to calculate your min sampling rate and then you can set it to that.